mindmeld/components/entity_resolver.py

```diff
@@ -42,60 +50,277 @@
+        self.normalizer = resource_loader.query_factory.normalize
 
-        if self._is_system_entity:
-            canonical_entities = []
-        else:
-            canonical_entities = self._resource_loader.get_entity_map(self.type).get(
-                "entities", []
-            )
-        self._no_canonical_entity_map = len(canonical_entities) == 0
+        # default configs useful for obtaining model cache path
+        for key, value in self.default_er_config.get("model_settings", {}).items():
+            er_config["model_settings"][key] = value
+        self.er_config = er_config
 
-    @property
-    def _es_index_name(self):
-        return EntityResolver.ES_SYNONYM_INDEX_PREFIX + "_" + self.type
+        self.dirty = False  # bool, True if exists any unsaved generated data that can be saved
+        self.ready = False  # bool, True if the model is fit by calling .fit()
 
     @property
-    def _use_text_rel(self):
-        return self._er_config["model_type"] == "text_relevance"
+    def default_er_config(self):
+        return {"model_settings": {}}
 
     @property
     def _use_double_metaphone(self):
-        return "double_metaphone" in self._er_config.get("phonetic_match_types", [])
+        return "double_metaphone" in self.er_config.get("model_settings", {}).get(
+            "phonetic_match_types", [])
+
+    @staticmethod
+    def _process_entity_map(entity_type, entity_map, normalizer=None, augment_lower_case=False,
+                            augment_title_case=False, augment_normalized=False):
+        """
+        Loads in the mapping.json file and stores the synonym mappings in a item_map
+            and a synonym_map
+
+        Args:
+            entity_type (str): The entity type associated with this entity resolver
+            entity_map (dict): The loaded mapping.json file for the given entity type
+            normalizer (callable): The normalizer to use, if provided, used to normalize synonyms
+            augment_lower_case (bool): If to extend the synonyms list with their lower-cased values
+            augment_title_case (bool): If to extend the synonyms list with their title-cased values
+            augment_normalized (bool): If to extend the synonyms list with their normalized values,
+                                        uses the provided normalizer
+        """
+
+        if augment_lower_case or augment_title_case or augment_normalized:
+            def _form_explorer(string):
+                _forms = []
+                if augment_lower_case:
+                    _forms.append(string.lower())
+                if augment_title_case:
+                    _forms.append(string.title())
+                if augment_normalized and normalizer:
+                    _forms.append(normalizer(string))
+                return _forms
+        else:
+            _form_explorer = None
+
+        item_map = {}
+        syn_map = {}
+        seen_ids = []
+        for item in entity_map.get("entities", []):
```

Comment:
I think this whole for-loop operation is a validation check. Lets pluck this out and put it into its own validation function

Reply:
I think the whole loop isn't validation check but the first few lines are. I think it  would be unnecessary to pluck it out as we need to iterate over all entities again which is anyways happening in this loop among other things.